### ISSUE TITLE
Shared base for rolling() and rolling_outlier_std()?

### ISSUE THREAD

**Issue by jbednar** [ioam/holoviews#1338]
Right now, `operations.timeseries.rolling()` and `operations.timeseries.rolling_outlier_std()` are separate classes inheriting from `ElementOperation`.  In practice, I've found that I typically want to plot the rolling output together with the outlier output, in part so that it's clear how the outliers have been defined (with respect to what baseline).  For that to work, I've had to make sure I pass the same parameters to both functions, which currently isn't possible in general because `rolling` supports more things than `rolling_outlier_std` does, even though they do share some parameters and use the same underlying pandas function.

To help make these match more easily, can we either combine these to share a single base class, either by having `rolling_outlier_std` inherit from `rolling`, or  both inherit from a new shared base class, or else have `timeseries.rolling_outlier_std` use `timeseries.rolling` rather than calling the pandas function?  Either way, we could set the same parameters once (at the class level) and know that they apply to both, or can pass a shared set of kwargs to both.

If we make this change, it would be good to do it in 1.7 because it represents a subtle API/behavior change that could be very confusing later.

**Comment by philippjfr** [comment]
A shared baseclass seems like the only sensible solution, because ``function`` makes no real sense on ``rolling_outlier_std``. The baseclass would only define these:

```python
    center = param.Boolean(default=True, doc="""
        Whether to set the x-coordinate at the center or right edge
        of the window.""")

    min_periods = param.Integer(default=None, doc="""
       Minimum number of observations in window required to have a
       value (otherwise result is NA).""")

    rolling_window = param.Integer(default=10, doc="""
        The window size over which to apply the function.""")

    window_type = param.ObjectSelector(default=None,
        objects=['boxcar', 'triang', 'blackman', 'hamming', 'bartlett',
                 'parzen', 'bohman', 'blackmanharris', 'nuttall',
                 'barthann', 'kaiser', 'gaussian', 'general_gaussian',
                 'slepian'], doc="The type of the window to apply")
```

**Comment by jbednar** [comment]
Fixed in #1344 .
